packages/components/pie-modal/src/index.ts

```diff
@@ -78,9 +85,10 @@ export class PieModal extends RtlMixin(LitElement) {
      */
     private _handleModalOpened () : void {
         disableBodyScroll(this);
-        //  We require this because toggling the prop `isOpen` itself won't
-        //  allow the dialog to open in the correct way (with the default background),
-        //  the method `showModal()` needs to be invoked.
+        if (this._dialog?.hasAttribute('open') || !this._dialog?.isConnected) {
```

Comment:
This line resolves these two errors in Storybook, and it's probably good defensive practice regardless.

<img src="https://github.com/justeattakeaway/pie/assets/26894168/d2751f89-47cc-4562-ada9-91b89b1b28b3" width="400px" />

Reply:
Is it worth clearing the open state before re-opening it? i.e close it first, would that clear these errors. Just a thought i'd check.